Move ElementIdentifier to NodeIdentifier

achristensen07 opened the pull request:
#### b794d867f83dfeaee9eb266a89a556f5973b9159
<pre>
Move ElementIdentifier to NodeIdentifier
<a href="https://rdar.apple.com/155807197">rdar://155807197</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=295909">https://bugs.webkit.org/show_bug.cgi?id=295909</a>

Reviewed by Ryosuke Niwa.

Document is both a Node and a ScriptExecutionContext which has an identifier()
function, so one of the identifier() functions needs to be renamed.  I chose
nodeIdentifier().

This will allow things like text nodes to be identified as well as elements.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::dispatchSystemPreviewActionEvent):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::~Element):
(WebCore::elementIdentifiersMap): Deleted.
(WebCore::Element::identifier const): Deleted.
(WebCore::Element::fromIdentifier): Deleted.
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementContext.h:
(WebCore::ElementContext::isSameElement const):
* Source/WebCore/dom/Node.cpp:
(WebCore::nodeIdentifiersMap):
(WebCore::Node::~Node):
(WebCore::Node::nodeIdentifier const):
(WebCore::Node::fromIdentifier):
* Source/WebCore/dom/Node.h:
* Source/WebCore/dom/NodeIdentifier.h: Renamed from Source/WebCore/dom/ElementIdentifier.h.
* Source/WebCore/dom/ViewTransition.cpp:
(WebCore::effectiveViewTransitionName):
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::handleClick):
* Source/WebCore/loader/EmptyClients.cpp:
* Source/WebCore/page/DragClient.h:
* Source/WebCore/page/DragController.cpp:
(WebCore::DragController::doSystemDrag):
* Source/WebCore/page/ElementTargetingController.cpp:
(WebCore::targetedElementInfo):
(WebCore::ElementTargetingController::topologicallySortElementsHelper):
(WebCore::ElementTargetingController::topologicallySortElements):
(WebCore::ElementTargetingController::findAllTargets):
(WebCore::ElementTargetingController::extractTargets):
(WebCore::ElementTargetingController::adjustVisibility):
(WebCore::ElementTargetingController::adjustVisibilityInRepeatedlyTargetedRegions):
(WebCore::ElementTargetingController::resetVisibilityAdjustments):
(WebCore::ElementTargetingController::snapshotIgnoringVisibilityAdjustment):
* Source/WebCore/page/ElementTargetingController.h:
* Source/WebCore/page/ElementTargetingTypes.h:
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/InteractionRegion.cpp:
(WebCore::interactionRegionForRenderedRegion):
* Source/WebCore/page/InteractionRegion.h:
(WebCore::operator==):
* Source/WebCore/page/WebKitNamespace.cpp:
(WebCore::WebKitNamespace::createNodeInfo):
* Source/WebCore/page/WebKitNodeInfo.cpp:
(WebCore::contentFrameIdentifier):
(WebCore::WebKitNodeInfo::WebKitNodeInfo):
* Source/WebCore/page/WebKitNodeInfo.h:
(WebCore::WebKitNodeInfo::create):
(WebCore::WebKitNodeInfo::nodeIdentifier const):
(WebCore::WebKitNodeInfo::elementIdentifier const): Deleted.
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::requestInteractiveModelElementAtPoint):
(WebCore::EventHandler::stageModeSessionDidUpdate):
(WebCore::EventHandler::stageModeSessionDidEnd):
* Source/WebCore/page/scrolling/ScrollAnchoringController.cpp:
(WebCore::ScrollAnchoringController::examineAnchorCandidate):
* Source/WebCore/page/scrolling/ScrollSnapOffsetsInfo.cpp:
(WebCore::updateSnapOffsetsForScrollableArea):
* Source/WebCore/page/scrolling/ScrollSnapOffsetsInfo.h:
* Source/WebCore/platform/ScrollSnapAnimatorState.cpp:
(WebCore::ScrollSnapAnimatorState::preserveCurrentTargetForAxis):
(WebCore::ScrollSnapAnimatorState::currentlySnappedBoxes const):
(WebCore::chooseBoxToResnapTo):
* Source/WebCore/platform/ScrollSnapAnimatorState.h:
* Source/WebCore/rendering/EventRegion.cpp:
(WebCore::EventRegionContext::uniteInteractionRegions):
(WebCore::EventRegionContext::shouldConsolidateInteractionRegion):
(WebCore::EventRegionContext::convertGuardContainersToInterationIfNeeded):
(WebCore::EventRegionContext::shrinkWrapInteractionRegions):
(WebCore::EventRegionContext::removeSuperfluousInteractionRegions):
* Source/WebCore/rendering/EventRegion.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::nodeIdentifier const):
(WebCore::Internals::isElementAlive const):
(WebCore::Internals::elementIdentifier const): Deleted.
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/JavaScriptEvaluationResult.cpp:
(WebKit::JavaScriptEvaluationResult::toVariant):
* Source/WebKit/Shared/JavaScriptEvaluationResult.serialization.in:
* Source/WebKit/Shared/NodeInfo.h:
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/API/APITargetedElementInfo.cpp:
(API::TargetedElementInfo::isSameElement const):
* Source/WebKit/UIProcess/API/APITargetedElementInfo.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKTextInputContext.mm:
(-[_WKTextInputContext hash]):
* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::startDrag):
* Source/WebKit/UIProcess/Model/StageModeInteractionState.h:
* Source/WebKit/UIProcess/PageClient.h:
(WebKit::PageClient::startDrag):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm:
(WebKit::interactionRegionGroupNameForRegion):
(WebKit::createInteractionRegionLayer):
(WebKit::updateLayersForInteractionRegions):
* Source/WebKit/UIProcess/SystemPreviewController.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::modelDragEnded):
(WebKit::WebPageProxy::stageModeSessionDidUpdate):
(WebKit::WebPageProxy::stageModeSessionDidEnd):
(WebKit::WebPageProxy::takeSnapshotForTargetedElement):
(WebKit::WebPageProxy::resetVisibilityAdjustmentsForTargetedElements):
(WebKit::WebPageProxy::adjustVisibilityForTargetedElements):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/DragDropInteractionState.h:
(WebKit::DragDropInteractionState::nodeIdentifier const):
(WebKit::DragDropInteractionState::elementIdentifier const): Deleted.
* Source/WebKit/UIProcess/ios/DragDropInteractionState.mm:
(WebKit::DragDropInteractionState::setElementIdentifier):
* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::didReceiveInteractiveModelElement):
(WebKit::PageClientImpl::startDrag):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _startDrag:item:nodeID:]):
(-[WKContentView cleanUpDragSourceSessionState]):
(-[WKContentView modelInteractionPanGestureDidUpdateWithPoint:]):
(-[WKContentView modelInteractionPanGestureDidEnd]):
(-[WKContentView didReceiveInteractiveModelElement:]):
(-[WKContentView _stageModeInfoForTesting]):
(-[WKContentView _startDrag:item:elementID:]): Deleted.
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::WebPageProxy::didReceiveInteractiveModelElement):
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::startDrag):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebDragClient.cpp:
(WebKit::WebDragClient::startDrag):
* Source/WebKit/WebProcess/WebCoreSupport/WebDragClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebDragClientMac.mm:
(WebKit::WebDragClient::startDrag):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::modelDragEnded):
(WebKit::WebPage::requestInteractiveModelElementAtPoint):
(WebKit::WebPage::stageModeSessionDidUpdate):
(WebKit::WebPage::stageModeSessionDidEnd):
(WebKit::WebPage::elementForContext const):
(WebKit::WebPage::contextForElement const):
(WebKit::WebPage::takeSnapshotForTargetedElement):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::textInputContextsInRect):
* Source/WebKitLegacy/mac/WebCoreSupport/WebDragClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebDragClient.mm:
(WebDragClient::startDrag):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/NodeInfo.mm:
(TestWebKitAPI::TEST(NodeInfo, Basic)):

Canonical link: <a href="https://commits.webkit.org/297382@main">https://commits.webkit.org/297382@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/223610c345ad355e7ca7a0705213a604b827bf11

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/111609 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/31276 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/21750 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/117644 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/61880 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/113571 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/31964 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/39861 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/84800 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/61880 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/114556 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/25514 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/100452 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65241 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/111042 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/24852 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/18587 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/61475 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/94898 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18656 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/120890 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/38662 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/28720 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/93702 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39040 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/96711 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93527 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38659 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/16453 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/34698 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17978 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/38551 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44035 "Built successfully") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/38214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/41550 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/39916 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/848deae936ef305a1e61f1e2831370629a2eee19. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47990

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d3e2452d4f6ed7e041a35480e753240ec7a08862. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47990

rniwa commented on Source/WebKit/UIProcess/WebPageProxy.h:
```diff
@@ -427,10 +427,10 @@ class Site;
 struct TextAnimationData;
 enum class ImageDecodingError : uint8_t;
-struct ElementIdentifierType;
+struct NodeIdentifierType;
```
Ditto.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/94db2505621f4a73f640c11297bc031b9822e2c1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47990

achristensen07 committed:
Move ElementIdentifier to NodeIdentifier
rdar://155807197
https://bugs.webkit.org/show_bug.cgi?id=295909

Reviewed by Ryosuke Niwa.

Document is both a Node and a ScriptExecutionContext which has an identifier()
function, so one of the identifier() functions needs to be renamed.  I chose
nodeIdentifier().

This will allow things like text nodes to be identified as well as elements.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::dispatchSystemPreviewActionEvent):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::~Element):
(WebCore::elementIdentifiersMap): Deleted.
(WebCore::Element::identifier const): Deleted.
(WebCore::Element::fromIdentifier): Deleted.
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementContext.h:
(WebCore::ElementContext::isSameElement const):
* Source/WebCore/dom/Node.cpp:
(WebCore::nodeIdentifiersMap):
(WebCore::Node::~Node):
(WebCore::Node::nodeIdentifier const):
(WebCore::Node::fromIdentifier):
* Source/WebCore/dom/Node.h:
* Source/WebCore/dom/NodeIdentifier.h: Renamed from Source/WebCore/dom/ElementIdentifier.h.
* Source/WebCore/dom/ViewTransition.cpp:
(WebCore::effectiveViewTransitionName):
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::HTMLAnchorElement::handleClick):
* Source/WebCore/loader/EmptyClients.cpp:
* Source/WebCore/page/DragClient.h:
* Source/WebCore/page/DragController.cpp:
(WebCore::DragController::doSystemDrag):
* Source/WebCore/page/ElementTargetingController.cpp:
(WebCore::targetedElementInfo):
(WebCore::ElementTargetingController::topologicallySortElementsHelper):
(WebCore::ElementTargetingController::topologicallySortElements):
(WebCore::ElementTargetingController::findAllTargets):
(WebCore::ElementTargetingController::extractTargets):
(WebCore::ElementTargetingController::adjustVisibility):
(WebCore::ElementTargetingController::adjustVisibilityInRepeatedlyTargetedRegions):
(WebCore::ElementTargetingController::resetVisibilityAdjustments):
(WebCore::ElementTargetingController::snapshotIgnoringVisibilityAdjustment):
* Source/WebCore/page/ElementTargetingController.h:
* Source/WebCore/page/ElementTargetingTypes.h:
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/InteractionRegion.cpp:
(WebCore::interactionRegionForRenderedRegion):
* Source/WebCore/page/InteractionRegion.h:
(WebCore::operator==):
* Source/WebCore/page/WebKitNamespace.cpp:
(WebCore::WebKitNamespace::createNodeInfo):
* Source/WebCore/page/WebKitNodeInfo.cpp:
(WebCore::contentFrameIdentifier):
(WebCore::WebKitNodeInfo::WebKitNodeInfo):
* Source/WebCore/page/WebKitNodeInfo.h:
(WebCore::WebKitNodeInfo::create):
(WebCore::WebKitNodeInfo::nodeIdentifier const):
(WebCore::WebKitNodeInfo::elementIdentifier const): Deleted.
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::requestInteractiveModelElementAtPoint):
(WebCore::EventHandler::stageModeSessionDidUpdate):
(WebCore::EventHandler::stageModeSessionDidEnd):
* Source/WebCore/page/scrolling/ScrollAnchoringController.cpp:
(WebCore::ScrollAnchoringController::examineAnchorCandidate):
* Source/WebCore/page/scrolling/ScrollSnapOffsetsInfo.cpp:
(WebCore::updateSnapOffsetsForScrollableArea):
* Source/WebCore/page/scrolling/ScrollSnapOffsetsInfo.h:
* Source/WebCore/platform/ScrollSnapAnimatorState.cpp:
(WebCore::ScrollSnapAnimatorState::preserveCurrentTargetForAxis):
(WebCore::ScrollSnapAnimatorState::currentlySnappedBoxes const):
(WebCore::chooseBoxToResnapTo):
* Source/WebCore/platform/ScrollSnapAnimatorState.h:
* Source/WebCore/rendering/EventRegion.cpp:
(WebCore::EventRegionContext::uniteInteractionRegions):
(WebCore::EventRegionContext::shouldConsolidateInteractionRegion):
(WebCore::EventRegionContext::convertGuardContainersToInterationIfNeeded):
(WebCore::EventRegionContext::shrinkWrapInteractionRegions):
(WebCore::EventRegionContext::removeSuperfluousInteractionRegions):
* Source/WebCore/rendering/EventRegion.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::nodeIdentifier const):
(WebCore::Internals::isElementAlive const):
(WebCore::Internals::elementIdentifier const): Deleted.
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/JavaScriptEvaluationResult.cpp:
(WebKit::JavaScriptEvaluationResult::toVariant):
* Source/WebKit/Shared/JavaScriptEvaluationResult.serialization.in:
* Source/WebKit/Shared/NodeInfo.h:
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/API/APITargetedElementInfo.cpp:
(API::TargetedElementInfo::isSameElement const):
* Source/WebKit/UIProcess/API/APITargetedElementInfo.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKTextInputContext.mm:
(-[_WKTextInputContext hash]):
* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::startDrag):
* Source/WebKit/UIProcess/Model/StageModeInteractionState.h:
* Source/WebKit/UIProcess/PageClient.h:
(WebKit::PageClient::startDrag):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm:
(WebKit::interactionRegionGroupNameForRegion):
(WebKit::createInteractionRegionLayer):
(WebKit::updateLayersForInteractionRegions):
* Source/WebKit/UIProcess/SystemPreviewController.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::modelDragEnded):
(WebKit::WebPageProxy::stageModeSessionDidUpdate):
(WebKit::WebPageProxy::stageModeSessionDidEnd):
(WebKit::WebPageProxy::takeSnapshotForTargetedElement):
(WebKit::WebPageProxy::resetVisibilityAdjustmentsForTargetedElements):
(WebKit::WebPageProxy::adjustVisibilityForTargetedElements):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/DragDropInteractionState.h:
(WebKit::DragDropInteractionState::nodeIdentifier const):
(WebKit::DragDropInteractionState::elementIdentifier const): Deleted.
* Source/WebKit/UIProcess/ios/DragDropInteractionState.mm:
(WebKit::DragDropInteractionState::setElementIdentifier):
* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::didReceiveInteractiveModelElement):
(WebKit::PageClientImpl::startDrag):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _startDrag:item:nodeID:]):
(-[WKContentView cleanUpDragSourceSessionState]):
(-[WKContentView modelInteractionPanGestureDidUpdateWithPoint:]):
(-[WKContentView modelInteractionPanGestureDidEnd]):
(-[WKContentView didReceiveInteractiveModelElement:]):
(-[WKContentView _stageModeInfoForTesting]):
(-[WKContentView _startDrag:item:elementID:]): Deleted.
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::WebPageProxy::didReceiveInteractiveModelElement):
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::startDrag):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebDragClient.cpp:
(WebKit::WebDragClient::startDrag):
* Source/WebKit/WebProcess/WebCoreSupport/WebDragClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebDragClientMac.mm:
(WebKit::WebDragClient::startDrag):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::modelDragEnded):
(WebKit::WebPage::requestInteractiveModelElementAtPoint):
(WebKit::WebPage::stageModeSessionDidUpdate):
(WebKit::WebPage::stageModeSessionDidEnd):
(WebKit::WebPage::elementForContext const):
(WebKit::WebPage::contextForElement const):
(WebKit::WebPage::takeSnapshotForTargetedElement):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::textInputContextsInRect):
* Source/WebKitLegacy/mac/WebCoreSupport/WebDragClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebDragClient.mm:
(WebDragClient::startDrag):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/NodeInfo.mm:
(TestWebKitAPI::TEST(NodeInfo, Basic)):

Canonical link: https://commits.webkit.org/297382@main
